Source/GSStoryboardTransform.m

```diff
@@ -0,0 +1,747 @@
+
+- (instancetype) initWithData: (NSData *)data
+{
+  self = [super init];
+  if (self != nil)
+    {
+      NSXMLDocument *xml = [[NSXMLDocument alloc] initWithData: data
+                                                       options: 0
+                                                         error: NULL];
+      
+      _scenesMap = [[NSMutableDictionary alloc] initWithCapacity: 10];
+      _controllerMap = [[NSMutableDictionary alloc] initWithCapacity: 10];
+      _documentsMap = [[NSMutableDictionary alloc] initWithCapacity: 10];
+      _identifierToSegueMap = [[NSMutableDictionary alloc] initWithCapacity: 10];
+      
+      [self processStoryboard: xml];
+      RELEASE(xml);
+    }
+  return self;
+}
+
+- (void) dealloc
+{
+  RELEASE(_initialViewControllerId);
+  RELEASE(_applicationSceneId);
+  RELEASE(_scenesMap);
+  RELEASE(_controllerMap);
+  RELEASE(_documentsMap);
+  RELEASE(_identifierToSegueMap);
+  [super dealloc];
+}
+
+- (NSString *) initialViewControllerId
+{
+  return _initialViewControllerId;
+}
+
+- (NSString *) applicationSceneId
+{
+  return _applicationSceneId;
+}
+
+- (NSDictionary *) scenesMap
+{
+  return _scenesMap;
+}
+
+- (NSDictionary *) controllerMap
+{
+  return _controllerMap;
+}
+
+- (NSDictionary *) documentsMap
+{
+  return _documentsMap;
+}
+
+- (NSDictionary *) identifierToSegueMap
+{
+  return _identifierToSegueMap;
+}
+
+- (NSMapTable *) segueMapForIdentifier: (NSString *)identifier
+{
+  return [_identifierToSegueMap objectForKey: identifier];
+}
+
+- (NSXMLElement *) createCustomObjectWithId: (NSString *)ident
+                                   userLabel: (NSString *)userLabel
+                                 customClass: (NSString *)className
+{
+  NSXMLElement *customObject =
+    [[NSXMLElement alloc] initWithName: @"customObject"];
+  NSXMLNode *idValue =
+    [NSXMLNode attributeWithName: @"id"
+                     stringValue: ident];
+  NSXMLNode *usrLabel =
+    [NSXMLNode attributeWithName: @"userLabel"
+                     stringValue: userLabel];
+  NSXMLNode *customCls =
+    [NSXMLNode attributeWithName: @"customClass"
+                     stringValue: className];
+  
+  [customObject addAttribute: idValue];
+  [customObject addAttribute: usrLabel];
+  [customObject addAttribute: customCls];
+
+  AUTORELEASE(customObject);
+  
+  return customObject;
+}
+
+- (NSData *) dataForIdentifier: (NSString *)identifier
+{
+  NSString *sceneId = [_controllerMap objectForKey: identifier];
+  NSXMLDocument *xml = [_scenesMap objectForKey: sceneId];
+  [self processSegues: xml
+                forId: identifier];
+  return [xml XMLData];
+}
+
+- (NSData *) dataForSceneId: (NSString *)sceneId
+{
+  NSXMLDocument *xml = [_scenesMap objectForKey: _applicationSceneId];
+  NSData *xmlData = [xml XMLData];
+  return xmlData;
+}
+
+- (NSData *) dataForApplicationScene
+{
+  return [self dataForSceneId: _applicationSceneId];
+}
+
+- (void) processStoryboard: (NSXMLDocument *)storyboardXml
```

Comment:
Now you have a separate class for the transformation, but the internal structure isn't any better. You still need to split out helper methods and add a few more FOR_IN usages and move the code around a bit.

Reply:
Yes, I know that.  Since everything is working and well tested, I would like to push to master and work on optimization as a separate task.

Reply:
All code has been broken out into manageable methods and FOR_IN is used universally